presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java

```diff
@@ -259,20 +269,37 @@ public void testHiveCoercionAvro()
     private void doTestHiveCoercion(HiveTableDefinition tableDefinition)
     {
         String tableName = mutableTableInstanceOf(tableDefinition).getNameInDatabase();
-
+        String dummyTableName = format(dummyTableNameFormat, tableName);
+        executeHiveQuery(format("DROP TABLE IF EXISTS %s", dummyTableName));
+        executeHiveQuery(getCreateDummyTableDDL(tableDefinition));
         executeHiveQuery(format("INSERT INTO TABLE %s " +
                         "PARTITION (id=1) " +
                         "VALUES" +
                         "(-1, 2, -3, 100, -101, 2323, 12345, 0.5)," +
                         "(1, -2, null, -100, 101, -2323, -12345, -1.5)",
-                tableName));
+                dummyTableName));
```

Comment:
I find the cost you pay in order to avoid duplication in the `VALUES` clause too high. By introducing the dummy table, a nontrivial amount of complexity is introduced to this test. It makes the test harder to read and understand.

Instead, I would manually write it out, and remove the dummy table:
```
                        "(-1, 2, -3, 100, -101, 2323, 12345, 0.5," +
                        "    named_struct(...)," +
                        "    array(named_struct(...))," +
                        "    map(..., named_struct(...)))," +
                        "(1, -2, null, -100, 101, -2323, -12345, -1.5," +
                        "    named_struct(...)," +
                        "    array(named_struct(...))," +
                        "    map(..., named_struct(...)))," +
```

Reply:
Just realized I have tried this before and it seems impossible.

    Unable to create temp file for insert values Expression of type TOK_FUNCTION not supported in insert/values

Reply:
There is actually no reason that these data is inserted with Hive. (Maybe there was at the time.) I changed it to use Presto, which offers the necessary SQL expressiveness.